### PR TITLE
fix(bazel): conditionally add custom toolchains

### PR DESCRIPTION
**Pull request by chingor13**
This allows autosynth to succeed at generating older commits of
googleapis and discovery_artifact_manager when the custom python
toolchains are not available.

Fixes: https://github.com/googleapis/java-compute/issues/146

### CODE REVIEW

**Comment by chingor13** [comment]
cc @vam-google

File: synthtool/gcp/gapic_bazel.py [hunk 1]
```diff
@@ -88,6 +88,9 @@ def _generate_code(
             api_definitions_repo = self._clone_googleapis()
             api_definitions_repo_name = "googleapis"
 
+        # conditionally apply custom toolchain
+        _check_for_custom_kokoro_bazel_rules(api_definitions_repo)
```
Comment (vam-google) [inline synthtool/gcp/gapic_bazel.py:92]: The naming is somewhat inaccurate. There are no custom kokoro rules, but there are either "custom build flags" or "custom toolchains".

File: synthtool/sources/git.py [hunk 2]
```diff
@@ -82,6 +82,7 @@ def clone(
             cmd = ["git", "clone", "--single-branch", url, dest]
             shell.run(cmd, check=True)
         else:
+            shell.run(["git", "reset", "--hard", "HEAD"], cwd=str(dest), check=True)
```
Comment (vam-google) [inline synthtool/sources/git.py:85]: Just curious, why is this needed now but was not needed before?

**Comment by SurferJeffAtGoogle** [comment]
Closing because this PR has been dormant for 90+ days.  Feel free to re-open if you intend to revise and merge.
